### ISSUE TITLE
Unable to dial to a wss multiaddr from browser

### ISSUE THREAD

**Issue by vaibhavmuchandi** [libp2p/js-libp2p#1783]
- **Version**:
libp2p:0.45.4
@libp2p/websockets:6.0.1

- **Platform**:
Chromuim 

- **Subsystem**:
Dialer, WebSockets

#### Description:
Trying to dial to a `wss` multiaddr from browser after disabling connection gater, unable to establish a connection to the node running in a nodejs environment.
I get the error `code: "ERR_TRANSPORT_DIAL_FAILED` and `WebSocket connection to 'wss://127.0.0.1:45477/p2p/12D3KooWK7e8BogXmgJCyEBXQMcqi1Pu2ZApW7q6Cdvqs6p7ZtdM` in the browser

**Comment by maschad** [comment]
Could you please share the snippets of how you have configured the nodes? Our [Getting Started](https://github.com/libp2p/js-libp2p/blob/master/doc/GETTING_STARTED.md) section outlines how to dial a node over [websockets](https://github.com/libp2p/js-libp2p-websockets) that may be of some help.

**Comment by vaibhavmuchandi** [comment]
Well, Im able to dial using websockets in the nodejs env. This is how I create a libp2p node on the nodejs and browser. I want my browser node to connect with my nodejs node without me having to set up a domain and a reverse proxy for my nodejs node. Im able to connect with a `circuit-relay-transport` but Im unable to dial the nodes directly from browser using their public IP.

```
       const httpServer = https.createServer({
         cert: fs.readFileSync('./test_certs/cert.pem'),
         key: fs.readFileSync('./test_certs/key.pem')
       })
        node = await createLibp2p({
          transports: [
            webSockets(       
             server: httpServer,
              websocket: {
               rejectUnauthorized: false
             }),
          ],
          connectionEncryption: [noise()],
          streamMuxers: [yamux(), mplex()],
          services: {
            identify: identifyService(),
            pubsub: gossipsub({ allowPublishToZeroPeers: true }),
            dht: kadDHT({
              allowQueryWithZeroPeers: true,
            }),
          },
          peerDiscovery: [
            bootstrap({
              list: bootstrappers,
            }),
          ],
        });
```
Browser:

```
createLibp2p({
          transports: [
            webSockets(),
            circuitRelayTransport({
              discoverRelays: 2,
            }),
          ],
          connectionEncryption: [noise()],
          streamMuxers: [yamux(), mplex()],
          services: {
            identify: identifyService(),
            pubsub: gossipsub({ allowPublishToZeroPeers: true }),
            dht: kadDHT({
              allowQueryWithZeroPeers: true,
              client: true
            }),
          },
          peerDiscovery: [
            bootstrap({
              list: bootstrappers,
            }),
          ],
          connectionGater: {
            denyDialMultiaddr: async () => false,
          },
        });
```

**Comment by vaibhavmuchandi** [comment]
Yes, I am trying to dial to a localhost address itself, never the less - I had my listen address set to `0.0.0.0:0`, I guess I forgot to revert back to the old config before posting here as I was testing some other things too. Sorry.

**Comment by achingbrain** [comment]
> Yes, I am trying to dial to a localhost address itself

Is your certificate self-signed?  Does your browser trust this certificate?

**Comment by vaibhavmuchandi** [comment]
Yup, self signed. Not sure if trusted or not. If it is required to manually verify the cert through browser then I havent done that.

**Comment by achingbrain** [comment]
Browsers will not accept self-signed certificates by default - you have to import them somehow first, either to the system keychain or the browser one (depending on browser).

**Comment by achingbrain** [comment]
If this is only going to localhost can you not just use insecure sockets and save SSL for production?

**Comment by vaibhavmuchandi** [comment]
Thank you, this was helpful.

**Comment by bradwbradw** [comment]
@vaibhavmuchandi did you get to solve this problem? I have a similar setup but keep getting websocket errors that won't go away.

`WebSocket connection to `wss://127.0.0.1:36852/p2p/12D3...LMN' failed:`

The error just ends with a colon with no indication of what the problem was.

I noticed you are setting up your own https server and passing that to `websockets()`. I understand if we set up a listen address like `/ip4/0.0.0.0/tcp/0/wss` it should encapsulate the websocket connection inside tcp, so we wouldn't need a separate server? Just wanting to validate my thinking on that also.

help / guidance appreciated. My goal is an https nodeJS relay that allows browser peers to connect and discover eachother
